lib/utility/error_monitor.rb

```diff
@@ -51,6 +51,7 @@ def initialize(
     def note_success
       @consecutive_error_count = 0
       @success_count += 1
+      @window_errors[@window_index] = false
```

Comment:
It's better to encapsulate set window error and increment window index into one method.
```ruby
def set_window_error(error)
  @window_errors[@window_index % @window_size] = error
  @window_index = (@window_index + 1) % @window_size
end
```